### PR TITLE
Fixed #195 issue

### LINKED ISSUE #195: Improving the structure and readability of api/main.py

### PR DESCRIPTION
**Pull request by OussamaERrafif**
Fixed #195 issue

### CODE REVIEW

**Review by MrTanoshii — CHANGES_REQUESTED** [review]
Hi @OussamaERrafif ,

Thank you for your first PR, please have a look at the requested changes and let me know if you need any help

File: src/honeybot/api/main.py [hunk 1]
```diff
@@ -25,6 +25,162 @@
...
+        self.required_modules = get_requirements()
+        self.time = time.time()
+
+        self.irc = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.is_listen_on = 1
+        dom = self.server_url.split(".")
+        self.domain = ".".join(dom[-2:])
+        self.sp_command = "hbot"
+        self.plugins = []
+        self.core_plugins = []
```
Comment (MrTanoshii) [inline src/honeybot/api/main.py:104]: `class BotCore` and its `__init__()` are duplicated 3 times, please remove the duplicates

First duplicate

File: src/honeybot/api/main.py [hunk 2]
```diff
@@ -25,6 +25,162 @@
...
```
Comment (MrTanoshii) [inline src/honeybot/api/main.py:181]: Second duplicate

File: src/honeybot/api/main.py [hunk 3]
```diff
@@ -25,6 +25,162 @@
...
+
+    # Rest of the code...
 class BotCore:
```
Comment (MrTanoshii) [inline src/honeybot/api/main.py:184]: Start of third duplicate until L208

File: src/honeybot/api/main.py [hunk 4]
```diff
@@ -241,26 +434,29 @@ def greet(self):
...
+                    self.quit()
+                except Exception as e:
+                    # logger.error(e)
+                    # print(e)
+                    # logger.debug("there was an error")
+                    # logger.debug(data)
+                    # logger.debug("!!")
+                    # logger.debug(line)
+                    # logger.debug("-" * 50)
+                    pass
```
Comment (MrTanoshii) [inline src/honeybot/api/main.py:459]: Too much indentation?

**Review by MrTanoshii — CHANGES_REQUESTED** [review]

**Comment by OussamaERrafif** [comment]
@MrTanoshii Thanks for noticing these errors

**Review by MrTanoshii — APPROVED** [review]

**Comment by MrTanoshii** [comment]
Thanks for the PR @OussamaERrafif

**Pull request merged**